probe-rs/src/probe/usb_util.rs

```diff
@@ -10,33 +11,72 @@ pub trait InterfaceExt {
 
 impl InterfaceExt for Interface {
     fn write_bulk(&self, endpoint: u8, buf: &[u8], timeout: Duration) -> io::Result<usize> {
-        let fut = async {
-            let comp = self.bulk_out(endpoint, buf.to_vec()).await;
-            comp.status.map_err(io::Error::other)?;
+        let mut endpoint = self
+            .endpoint::<Bulk, Out>(endpoint)
+            .map_err(io::Error::from)?;
 
-            let n = comp.data.actual_length();
-            Ok(n)
+        if buf.is_empty() {
+            return Ok(0);
+        }
```

Comment:
An OUT transfer of zero bytes sends a zero-length packet. Don't know if any probes rely on that, but it's observable by the device, sometimes used as a delimiter, and not a no-op.

Reply:
Interesting! I will remove the checks from both write and read in the next commit.